src/build-data/compile_commands.json.in

```diff
@@ -1,22 +1,9 @@
 [
-%{for lib_build_info}
-    { "directory": "%{abs_root_dir}",
-      "command": "%{cxx} %{lib_flags} %{cxx_abi_flags} %{cc_lang_flags} %{os_feature_macros} %{isa_flags} %{include_paths} %{dash_c} %{src} %{dash_o}%{obj}",
-      "file": "%{src}"
-    },
-%{endfor}
-
-%{for cli_build_info}
-    { "directory": "%{abs_root_dir}",
-      "command": "%{cxx} %{cxx_abi_flags} %{cc_lang_flags} %{os_feature_macros} %{isa_flags} %{include_paths} %{dash_c} %{src} %{dash_o}%{obj}",
-      "file": "%{src}"
-    },
-%{endfor}
-
-%{for test_build_info}
-    { "directory": "%{abs_root_dir}",
-      "command": "%{cxx} %{cxx_abi_flags} %{cc_lang_flags} %{os_feature_macros} %{isa_flags} %{include_paths} %{dash_c} %{src} %{dash_o}%{obj}",
-      "file": "%{src}"
-    },
+%{for all_build_info}
```

Comment:
For consistency with `makefile.in`, would it not make more sense to keep the separation of `lib_build_info`/`cli_build_info`/`test_build_info`?

Would also avoid needing the new `build_info_lib_flags` variable.

Reply:
I'd love to. But then the `%{omitlast ,}` voodoo wouldn't work as smoothly. Because the mentioned build info variables are optional, we may or may not put a comma behind any of their loop blocks. That would complicate the template quite a bit.

Reply:
I changed this as discussed out-of-band to @lieser's idea. By putting the `lib_build_info` last, we should be fairly safe as building anything in Botan without the library doesn't make sense.